chain/chunks/src/lib.rs

```diff
@@ -1117,6 +1114,16 @@ impl ShardsManager {
             }
         };
 
+        // We must check the protocol version every time, since a new value
+        // could be passed to the function, whereas the signature check is intrinsic
+        // to the header, thus only needs to happen exactly once.
+        let partial_encoded_chunk = partial_encoded_chunk.unwrap();
+        if !partial_encoded_chunk.header.version_range().contains(protocol_version) {
```

Comment:
actually what is the purpose of this check? Does this mean that version v1 cannot be processed if you pass a version that is higher than `SHARD_CHUNK_HEADER_UPGRADE_VERSION`?

Reply:
This check prevents accepting a chunk which uses an old version after the network has collectively decided to upgrade. It should not prevent accepting an old chunk for an old block because you would pass the function the protocol version associated with that block's epoch.

It's also not just about v1 vs v2, we may introduce new versions in the future, and again need to make sure each version is only accepted during the epochs where they were valid.